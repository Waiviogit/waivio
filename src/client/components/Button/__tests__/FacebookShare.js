import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';

import FacebookShare from '../FacebookShare';

describe('<FacebookShare />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      url: '',
    };

    wrapper = mount(<FacebookShare {...props} />);
    act(() => {
      wrapper.update();
    });
  });

  it('Should include ShareButton in className in tag a ', () => {
    expect(wrapper.find('a').props().className).toBe('ShareButton');
  });

  it('Should include ShareButton__contents ShareButton__contents__facebook in className in tag div ', () => {
    expect(wrapper.find('div').props().className).toBe(
      'ShareButton__contents ShareButton__contents__facebook',
    );
  });

  it('Should include iconfont icon-facebook ShareButton__icon in className in tag i ', () => {
    expect(wrapper.find('i').props().className).toBe('iconfont icon-facebook ShareButton__icon');
  });

  it('should render without exploding', () => {
    wrapper = shallow(<FacebookShare {...props} />);
    act(() => {
      wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have defaultProps', () => {
    const defaultProps = {
      url: '',
    };

    expect(FacebookShare.defaultProps).toEqual(defaultProps);
  });

  it('Should have smth with the same text as url prop', () => {
    const url = wrapper.find('FacebookShare');

    expect(url.text()).toBe('');
  });

  it('Should not to be undefined component', () => {
    expect(wrapper).not.toBeUndefined();
  });
});
