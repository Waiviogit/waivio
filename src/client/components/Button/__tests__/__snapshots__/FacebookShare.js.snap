// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacebookShare /> should render without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FacebookShare
    url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="ShareButton__contents ShareButton__contents__facebook"
      >
        <i
          className="iconfont icon-facebook ShareButton__icon"
        />
      </div>,
      "className": "ShareButton",
      "href": "https://facebook.com/sharer.php?u=",
      "rel": "noopener noreferrer",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="iconfont icon-facebook ShareButton__icon"
        />,
        "className": "ShareButton__contents ShareButton__contents__facebook",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "iconfont icon-facebook ShareButton__icon",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ShareButton__contents ShareButton__contents__facebook"
        >
          <i
            className="iconfont icon-facebook ShareButton__icon"
          />
        </div>,
        "className": "ShareButton",
        "href": "https://facebook.com/sharer.php?u=",
        "rel": "noopener noreferrer",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="iconfont icon-facebook ShareButton__icon"
          />,
          "className": "ShareButton__contents ShareButton__contents__facebook",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconfont icon-facebook ShareButton__icon",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
